hpack/table.py

```diff
@@ -170,14 +170,17 @@ def search(self, name, value):
             - ``(index, name, None)`` for partial matches on name only.
             - ``(index, name, value)`` for perfect matches.
         """
-        offset = HeaderTable.STATIC_TABLE_LENGTH + 1
         partial = None
-        for (i, (n, v)) in enumerate(HeaderTable.STATIC_TABLE):
-            if n == name:
-                if v == value:
-                    return i + 1, n, v
-                elif partial is None:
-                    partial = (i + 1, n, None)
+
+        header_name_search_result = HeaderTable.STATIC_TABLE_MAPPING.get(name)
+        if header_name_search_result is not None:
```

Comment:
Sorry! I was hoping you hadn't read it because what I wrote was the complete opposite intent of what I meant to say. I really need to audit my comments before I snap-post. :stuck_out_tongue: